### DISCUSSION TITLE
CRXJS w/ Sandbox Page That Loads Remote React Script -- Best Practice?

### DISCUSSION THREAD

**Discussion by Jgordon-pencilwrench** [crxjs/chrome-extension-tools#837] [Q&A] [answered]
Asking for best practices here:

# The Goal:
I want to avoid going through the review process when I make changes to my extension, so I've decided to explore using a sandboxed page that loads a react app from my CDN.

# The Implementation Plan
The user will interact with the page, then the sandboxed page will post a message for my extension to handle. I think it will work, but I'm looking for suggestions on how to handle developing this app.

## The Issues I see:

1. CRXJS doesn't look like it works with sandboxed pages for local development. I'd need to create a second page that isn't sandboxed to use during development
2. I probably have to develop two projects: one for the extension, one for the sandboxed code.
3. *SOLVED* :  It's a react-vite app I'll be loading. I'm wondering if I'll run into issues with hashing the scripts and css files (though I could turn off hashing in my vite config at the risk of caching stale code). -- Set the cache expiration in the headers to expire at x time.

Have other folks done this before (I'm assuming yesssss)? What is the ideal development flow for this kind of thing? I'm pretty sure this would pass Google's review, but would love to hear other people's experience.

**Comment by rachelslurs** [comment]
I recognize this was over a year ago, but I'm wondering if you had any luck figuring this out!

Thank you!

**Reply by Jgordon-pencilwrench** [reply, marked as answer]
I did. I turns out that it's okay to load a page from a remote host. I load the page in a popup window. The issue that Google is concerned about is downloading code from an external host and running it within the chrome extension environment via `evaluate`. If you're just loading an external website, you can go ahead and do that.
